### PR TITLE
[do not merge yet] JBPM-6592: Prepare test framework for image testing

### PR DESCRIPTION
**Pull request by sutaakar**

### CODE REVIEW

**Comment by sutaakar** [comment]
retest this please

**Comment by sutaakar** [comment]
Adjusted Image stream wait condition in ImageTestBase to verify availability of all image streams.

File: framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ImageImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public ImageImpl(OpenShiftClient client, String projectName, ImageStream imageStream) {
+        this.client = client;
+        this.projectName = projectName;
+        this.imageStream = imageStream;
+    }
+
+    @Override
+    public String getImageReference() {
+        ImageStreamTag imageStreamTag = client.imageStreamTags().inNamespace(projectName).withName(imageStream.getImageStreamName()).get();
```
Comment (jhrcek) [inline framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ImageImpl.java]: I'm wondering what's the difference between this and the same expression in ProjectImpl#getImage. There you've got the nullcheck, but here not. Can't this lead to NPE here?

Comment (sutaakar) [reply framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/resource/impl/ImageImpl.java]: @jhrcek Initial idea is that framework resource (Image, Pod, Service) represents existing OpenShift resource. So ProjectImpl#getImage checks that the image exists and then resource is returned, he we can assume that the image exists (as we already have its framework representation).

**Comment by sutaakar** [comment]
Will keep this PR opened until the template situation is clarified.

**Comment by hradecek** [comment]
retest this

**Comment by sutaakar** [comment]
This functionality should be tested by image tests in jboss-kie-modules

**Pull request closed without merging**